internal/server/server.go

```diff
@@ -35,27 +36,37 @@ func (s *Server) Init() *Server {
 	}
 
 	s.router.Handle("/metrics", metrics.Handler())
-	s.router.HandleFunc("GET    /health", s.health)
-	s.router.HandleFunc("GET    /v1/sources/{source}/data/{path...}", authenticationMiddleware(s.db, s.v1SourcesDataGet))
-	s.router.HandleFunc("POST   /v1/sources/{source}/data/{path...}", authenticationMiddleware(s.db, s.v1SourcesDataPut))
-	s.router.HandleFunc("PUT    /v1/sources/{source}/data/{path...}", authenticationMiddleware(s.db, s.v1SourcesDataPut))
-	s.router.HandleFunc("DELETE /v1/sources/{source}/data/{path...}", authenticationMiddleware(s.db, s.v1SourcesDataDelete))
-	s.router.HandleFunc("GET    /v1/sources", authenticationMiddleware(s.db, s.v1SourcesList))
-	s.router.HandleFunc("PUT    /v1/sources/{source}", authenticationMiddleware(s.db, s.v1SourcesPut))
-	s.router.HandleFunc("GET    /v1/sources/{source}", authenticationMiddleware(s.db, s.v1SourcesGet))
-	s.router.HandleFunc("DELETE /v1/sources/{source}", authenticationMiddleware(s.db, s.v1SourcesDelete))
-	s.router.HandleFunc("GET    /v1/bundles", authenticationMiddleware(s.db, s.v1BundlesList))
-	s.router.HandleFunc("PUT    /v1/bundles/{bundle}", authenticationMiddleware(s.db, s.v1BundlesPut))
-	s.router.HandleFunc("GET    /v1/bundles/{bundle}", authenticationMiddleware(s.db, s.v1BundlesGet))
-	s.router.HandleFunc("DELETE /v1/bundles/{bundle}", authenticationMiddleware(s.db, s.v1BundlesDelete))
-	s.router.HandleFunc("GET    /v1/stacks", authenticationMiddleware(s.db, s.v1StacksList))
-	s.router.HandleFunc("PUT    /v1/stacks/{stack}", authenticationMiddleware(s.db, s.v1StacksPut))
-	s.router.HandleFunc("GET    /v1/stacks/{stack}", authenticationMiddleware(s.db, s.v1StacksGet))
-	s.router.HandleFunc("DELETE /v1/stacks/{stack}", authenticationMiddleware(s.db, s.v1StacksDelete))
-	s.router.HandleFunc("GET    /v1/secrets", authenticationMiddleware(s.db, s.v1SecretsList))
-	s.router.HandleFunc("PUT    /v1/secrets/{secret}", authenticationMiddleware(s.db, s.v1SecretsPut))
-	s.router.HandleFunc("GET    /v1/secrets/{secret}", authenticationMiddleware(s.db, s.v1SecretsGet))
-	s.router.HandleFunc("DELETE /v1/secrets/{secret}", authenticationMiddleware(s.db, s.v1SecretsDelete))
+	s.router.HandleFunc("GET /health", s.health)
+
+	base := chain.New(authenticationMiddleware(s.db))
+	setup := func(method, pattern string, hndl http.HandlerFunc) {
+		s.router.Handle(method+" "+pattern, append(base, metrics.InstrumentHandler(pattern)).ThenFunc(hndl))
+	}
+
+	setup("GET", "/v1/sources/{source}/data/{path...}", s.v1SourcesDataGet)
```

Comment:
I like the extra space here. It breaks up the endpoint groupings so it's easier to scan visually.